CMakeLists.txt

```diff
@@ -1,84 +1,79 @@
-cmake_minimum_required(VERSION 3.8)
-project(LittleTypeLibrary)
-
-if(MSVC)
-        add_compile_options(
-            /Zc:preprocessor
-            /permissive-
-            /d2cgsummary
-            /Wv:18
-            /bigobj
-            /MT)
-endif(MSVC)
-
-set(CMAKE_CXX_STANDARD 17)
-
-set(HEADERS_RANGE
-    include/ltl/Range/Map.h
-    include/ltl/Range/Zip.h
-    include/ltl/Range/Join.h
-    include/ltl/Range/Split.h
-    include/ltl/Range/Range.h
-    include/ltl/Range/Taker.h
-    include/ltl/Range/Value.h
-    include/ltl/Range/Filter.h
-    include/ltl/Range/actions.h
-    include/ltl/Range/Reverse.h
-    include/ltl/Range/AsPointer.h
-    include/ltl/Range/enumerate.h
-    include/ltl/Range/DefaultView.h
-    include/ltl/Range/BaseIterator.h
-    include/ltl/Range/NullableFunction.h
-    )
-
-set(HEADERS
-    include/ltl/ltl.h
-    include/lpl/lpl.h
-    include/ltl/algos.h
-    include/ltl/crtp.h
-    include/ltl/Tuple.h
-    include/ltl/stream.h
-    include/ltl/invoke.h
-    include/ltl/traits.h
-    include/ltl/concept.h
-    include/ltl/expected.h
-    include/ltl/operator.h
-    include/ltl/condition.h
-    include/ltl/StrongType.h
-    include/ltl/TypedTuple.h
-    include/ltl/functional.h
-    include/ltl/tuple_algos.h
-    include/ltl/movable_any.h
-    include/ltl/VariantUtils.h
-    include/ltl/optional_type.h
-    ${HEADERS_RANGE}
-    )
```

Comment:
I saw you remove all my files from CMake. Does the IDE will check and parse the lib?

Reply:
Depends on the IDE.

I assume you are using VS. In VS's case, at the moment there is no way to associate files to `INTERFACE` targets that would make headers show up under the `Header Files` option in the Solution Explorer. This is fixed in CMake 3.19, but that's not released yet.  
Unfortunately there is no clean solution to this yet, the only thing you can do is generate a dummy.cpp file in the build directory and associate the headers files to that.

Reply:
Ahaha okay, in the worst case, I will add headers as I did before :).

I am using Qt Creator with MSVC compiler as default. But I have Mingw 7.3 and 8.1, and clang. I am developping this library at home
At work, I don't know which version of CMake we are using. I hope CMake 3.14 is okay. Else I will have to make some modifications .

Thanks for your time, I am going to continue to read the PR :).

Reply:
Your CMake version wasn't specified correctly before anyway. The earliest CMake version this library really needs is [3.11](https://cmake.org/cmake/help/latest/release/3.11.html#commands) if you use the old style of defining the `NAMELINK` component, but honestly this is a header only library without build artifacts, so even that is not really necessary.